Tangem/App/ViewModels/WalletModel/WalletModel+Balance.swift

```diff
@@ -32,13 +32,32 @@ extension WalletModel {
         availableBalance.fiat
     }
 
+    var isSuccessfullyLoaded: Bool {
+        let isStakingSuccessfullyLoaded = stakingManager?.state.isSuccessfullyLoaded ?? true
+        return state.isSuccessfullyLoaded && isStakingSuccessfullyLoaded
+    }
+
     var totalBalance: Balance {
         let cryptoBalance: Decimal? = {
             switch (availableBalance.crypto, stakedBalance.crypto) {
-            case (.none, _): nil
-            // What we have to do if we have only blocked balance?
-            case (.some(let available), .none): available
-            case (.some(let available), .some(let blocked)): available + blocked
+            case (.none, _):
+                return nil
+            case (.some(let available), .none):
+                // staking unsupported
+                guard let stakingManager else {
+                    return available
+                }
+
+                // no staked balance
+                if stakingManager.state.isSuccessfullyLoaded {
+                    return available
+                }
+
+                // staking error
+                return nil
```

Comment:
Вообще получается что для `staking unsupported` будет 
`stakedBalance.crypto == .none`

как вариант можно так 
```suggestion
            // Staking unsupported
            case (.some(let available), .none) where stakingManagerState == .notEnabled: available
            // Staking balances loading failed
            case (.some(let available), .none) where !stakingManagerState.state.isSuccessfullyLoaded: nil
```

Reply:
блин, только не пойму, по факту если `isSuccessfullyLoaded == true` мы должны попасть в последний кейс, так что тут проверять нет смысла это)

Reply:
да хз, кажется более читабельно, когда по кейсам сгруппировано

Reply:
ну тут тип доп условие почему именно `stakedBalance.crypto == .none`
ну можно так оставить, да